components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/DeviceManagementServiceProviderImpl.java

```diff
@@ -596,7 +596,48 @@ public int getDeviceCount() throws DeviceManagementException {
             int deviceCount = this.deviceDAO.getDeviceCount();
             return deviceCount;
         } catch (DeviceManagementDAOException e) {
-            throw new DeviceManagementException("Error occurred while obtaining devices all devices", e);
+            log.error("Error occurred while counting devices", e);
+            throw new DeviceManagementException("Error occurred while counting devices", e);
+        }
+    }
+
+    @Override
+    public List<Device> getDevicesByName(String deviceName, int tenantId) throws DeviceManagementException {
+        List<Device> devicesOfUser = new ArrayList<Device>();
+        List<org.wso2.carbon.device.mgt.core.dto.Device> devicesList;
+        Device convertedDevice;
+        DeviceIdentifier deviceIdentifier;
+        DeviceManager dms;
+        Device dmsDevice;
+        org.wso2.carbon.device.mgt.core.dto.Device device;
```

Comment:
Noted and fixed in commit [1].
[1] https://github.com/wso2/carbon-device-mgt/commit/60960d80b4cfa1e7bf16c3fffb10f9295e2185ae